packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -620,6 +620,16 @@ class EditableText extends StatefulWidget {
   /// context, the English phrase will be on the right and the Hebrew phrase on
   /// its left.
   ///
+  /// When text from an LTR language is entered into an RTL field, or RTL text
+  /// is entered into an LTR field,
+  /// [LRM](https://en.wikipedia.org/wiki/Left-to-right_mark) or
+  /// [RLM](https://en.wikipedia.org/wiki/Right-to-left_mark) characters will be
+  /// inserted alongside whitespace characters, respectively. For example, an
+  /// LRM character will be inserted when an English keyboard is used to enter a
```

Comment:
It is not the type of keyboard, but rather the mark inserted is based on the directionality of the immediately previous run of text. This ensures the whitespace is treated as part of the same directionality run instead of defaulting to the underlying direction of the field.

Reply:
@GaryQian Alright I tried to clean this paragraph up a little bit, and I removed this example sentence entirely.  I think it was making things even more confusing by introducing uncertainty.  They first sentence says it as clearly as possible: typing LTR text into an RTL field will produce LRM characters if it includes spaces.